packages/plugin-core/src/test/suite-integ/PreviewLinkHandler.test.ts

```diff
@@ -1,247 +1,416 @@
 
-            // with urlencoded parts (space as %20)
-            const withURLEncode = handler.vaultlessAssetPath({
+        describe("AND the link is to a block anchor", () => {
+          test("THEN the note is opened at that block", async () => {
+            const handler = new PreviewLinkHandler(
+              ExtensionProvider.getExtension()
+            );
+            const out = await handler.onLinkClicked({
               data: {
-                href: "vscode-webview://e380a62c-2dea-46a8-ae1e-a34868c9719e/assets/file%20with%20space.pdf",
-                id: "foo",
+                href: "vscode-webview://76b3da02-f902-4652-b6a8-746551d032ce/test-id#^nihil",
+                id: note.id,
               },
-              wsRoot,
             });
-            expect(withURLEncode !== undefined).toBeTruthy();
+            expect(out).toEqual(LinkType.WIKI);
             expect(
-              await AssertUtils.assertInString({
-                body: withURLEncode as string,
-                match: [path.join("assets", "file with space.pdf")],
-              })
+              VSCodeUtils.getActiveTextEditor()?.document.fileName.endsWith(
+                "target.md"
+              )
             ).toBeTruthy();
-
-            done();
-          },
+            expect(
+              VSCodeUtils.getActiveTextEditor()?.selection.active.line
+            ).toEqual(13);
+          });
         });
-      });
-    });
-  });
 
-  // this test block passes because it doesn't wait for the actual test to run.
-  // the actual test does not pass.
-  describe.skip(`handleLink`, () => {
-    describe(`LinkType.ASSET`, () => {
-      describe(`WHEN valid href`, () => {
-        let assetOpenerStub: any;
-
-        beforeEach(async () => {
-          const factory = TestNoteFactory.defaultUnitTestFactory();
+        describe("AND if the link is to a missing note", () => {
+          test("THEN nothing happens", async () => {
+            const handler = new PreviewLinkHandler(
+              ExtensionProvider.getExtension()
+            );
+            const openWithDefaultApp = sinon.stub(
+              ShowPreviewAssetOpener,
+              "openWithDefaultApp"
+            );
+            const out = await handler.onLinkClicked({
+              data: {
+                href: "vscode-webview://76b3da02-f902-4652-b6a8-746551d032ce/does-not-exist",
+                id: note.id,
+              },
+            });
+            expect(out).toEqual(LinkType.UNKNOWN);
+            expect(openWithDefaultApp.called).toBeFalsy();
+          });
+        });
 
-          await factory.createForFName("foo");
+        describe("AND the link is ambiguous", () => {
+          test("THEN it prompts for a note", async () => {
+            const { engine } = ExtensionProvider.getDWorkspace();
+            const showChooseNote = sinon
+              .stub(QuickPickUtil, "showChooseNote")
+              .returns(Promise.resolve(engine.notes["reprehenderit"]));
```

Comment:
can we store these `reprehenderit` strings into a const variable and reuse?

Reply:
Do you mean just adding `const showChooseNoteTargetId = "reprehenderit";` to the top of this test file? It is only used here and when creating the note, and the test would fail if someone changed something so we couldn't find the note. I don't think there's really anything to reuse, I'd rather leave this as-is.

Reply:
It's more of a nit, but yes - just for the point of consolidating it between the note creation step and the validation step.  It's more for readability instead of reusability - it makes it a little bit easier to understand what we're trying to verify.

Reply:
Got it, I'll go ahead and refactor that then. I can actually grab the entire note, so it just uses the right note instead of looking it up.